### PR TITLE
IR-3663 video controls size and responsiveness

### PR DESCRIPTION
**Pull request by SamMazerIR**
## Summary
fixing the mediacontrols button scaling and reactivity (play/pause button icons update). Also managed to set scale information properly depending video canvas size and transform scale (chooses smaller of combined size for controls scale)

image
are you not entertained?

## References
[https://tsu.atlassian.net/browse/IR-3663](https://tsu.atlassian.net/browse/IR-3663)

## QA Steps
put a video in your scene, and update the transform's scale and the video canvas size as you see fit. The media controls in the published scene should be roughly 10% of the size of the smaller dimension of width or height

### CODE REVIEW

File: packages/client-core/src/systems/ui/MediaControlsUI.tsx [hunk 1]
```diff
@@ -45,21 +59,29 @@ type MediaControlsProps = {
 const MediaControlsView = (props: MediaControlsProps) => {
   const mediaComponent = useHookstate(getMutableComponent(props.entity, MediaComponent))
 
+  const controlsPercent = '10%'
```
Comment (SYBIOTE) [inline packages/client-core/src/systems/ui/MediaControlsUI.tsx]: i think these belong in a common file, as proper constants

File: packages/client-core/src/systems/ui/MediaControlsUI.tsx [hunk 2]
```diff
@@ -87,7 +108,11 @@ const MediaControlsView = (props: MediaControlsProps) => {
             background-color: grey;
         }`}
         </style>
-        {mediaComponent.paused.value ? <PlayArrow style={{ fill: 'white' }} /> : <Pause style={{ fill: 'white' }} />}
+        {mediaComponent.paused.value ? (
+          <PlayArrow style={{ fill: 'white', width: `100%`, height: `100%` }} />
```
Comment (SYBIOTE) [inline packages/client-core/src/systems/ui/MediaControlsUI.tsx]: its the client so not relevant for now, but we need to eventually write these in tailwind, is it possible to write them in tailwind now

**Review by SYBIOTE — CHANGES_REQUESTED** [review]

File: packages/client-core/src/systems/ui/MediaControlsUI.tsx (hunk 1 above)
Comment (SamMazerIR) [reply packages/client-core/src/systems/ui/MediaControlsUI.tsx]: replaced these variables with constant values direct in the styles

File: packages/client-core/src/systems/ui/MediaControlsUI.tsx (hunk 2 above)
Comment (SamMazerIR) [reply packages/client-core/src/systems/ui/MediaControlsUI.tsx]: as you saw in call, tailwind is not currently supported here. added a todo

File: packages/client-core/src/systems/ui/MediaControlsUI.tsx [hunk 3]
```diff
@@ -46,20 +60,26 @@ const MediaControlsView = (props: MediaControlsProps) => {
...
 
   const buttonClick = () => {
-    const mediaElement = getComponent(props.entity, MediaElementComponent)
-    if (!mediaElement) return
-    mediaElement.element.paused ? mediaElement.element.play() : mediaElement.element.pause()
+    //early out if the mediaElement is null
+    if (!hasComponent(props.entity, MediaElementComponent)) return
+
+    const isPaused = mediaComponent.paused.value
+    isPaused ? mediaComponent.paused.set(false) : mediaComponent.paused.set(true)
```
Comment (mikeplascdev) [inline packages/client-core/src/systems/ui/MediaControlsUI.tsx]:
```suggestion
    mediaComponent.paused.set(!isPaused)

File: packages/client-core/src/systems/ui/MediaControlsUI.tsx [hunk 4]
```diff
@@ -87,7 +106,11 @@ const MediaControlsView = (props: MediaControlsProps) => {
             background-color: grey;
         }`}
         </style>
-        {mediaComponent.paused.value ? <PlayArrow style={{ fill: 'white' }} /> : <Pause style={{ fill: 'white' }} />}
+        {mediaComponent.paused.value ? (
+          <PlayArrow style={{ fill: 'white', width: `100%`, height: `100%` }} />
+        ) : (
+          <Pause style={{ fill: 'white', width: `100%`, height: `100%` }} />
+        )}
```
Comment (mikeplascdev) [inline packages/client-core/src/systems/ui/MediaControlsUI.tsx]:
```suggestion
        const mediaStyles = { fill: 'white', width: `100%`, height: `100%` };
        {mediaComponent.paused.value && (
          <PlayArrow style={mediaStyles} />
        )}
        {!mediaComponent.paused.value && (
          <Pause style={mediaStyles} />
        )}

File: packages/client-core/src/systems/ui/MediaControlsUI.tsx [hunk 5]
```diff
@@ -87,7 +106,13 @@ const MediaControlsView = (props: MediaControlsProps) => {
             background-color: grey;
         }`}
         </style>
-        {mediaComponent.paused.value ? <PlayArrow style={{ fill: 'white' }} /> : <Pause style={{ fill: 'white' }} />}
+        const mediaStyles = { fill: 'white', width: `100%`, height: `100%` };
```
Comment (mikeplascdev) [inline packages/client-core/src/systems/ui/MediaControlsUI.tsx]: we need to move this constant to the top after the const `mediaComponent` declaration line 61

Comment (SamMazerIR) [reply packages/client-core/src/systems/ui/MediaControlsUI.tsx]: done

**Review by SYBIOTE — APPROVED** [review]
looks good

**Review by mikeplascdev — APPROVED** [review]
LGTM

**Pull request merged**